arbnode/dataposter/slice/slicestorage.go

```diff
@@ -71,6 +71,8 @@ func (s *Storage[Item]) Put(_ context.Context, index uint64, prevItem *Item, new
 		if queueIdx > len(s.queue) {
 			return fmt.Errorf("attempted to set out-of-bounds index %v in queue starting at %v of length %v", index, s.firstNonce, len(s.queue))
 		}
+		// TODO: come up with a better way of comparing prev/new values,
+		// we shouldn't use cmp.Diff outside tests.
 		if diff := cmp.Diff(prevItem, s.queue[queueIdx]); diff != "" {
```

Comment:
FYI the original code did intentionally use pointer comparisons, because that worked with the batch poster's design. But I agree it isn't ideal. We should either RLP encode both and compare them here, or we could use reflect.DeepEqual.

Reply:
Done.